packages/server/src/integrations/arangodb.js

```diff
@@ -0,0 +1,85 @@
+const { Database, aql } = require("arangojs")
+const { FIELD_TYPES, QUERY_TYPES } = require("./Integration")
+
+const SCHEMA = {
+    docs: "https://github.com/arangodb/arangojs",
+    datasource: {
+        url: {
+            type: FIELD_TYPES.STRING,
+            default: "http://localhost:8529",
+            required: true,
+        },
+        username: {
+            type: FIELD_TYPES.STRING,
+            default: "root",
+            required: true,
+        },
+        password: {
+            type: FIELD_TYPES.PASSWORD,
+            required: true,
+        },
+        databaseName: {
+            type: FIELD_TYPES.STRING,
+            default: "_system",
+            required: true,
+        },
+        collection: {
+            type: FIELD_TYPES.STRING,
+            required: true,
+        },
+    },
+    query: {
+        read: {
+            type: QUERY_TYPES.SQL,
+        },
+        create: {
+            type: QUERY_TYPES.JSON,
+        },
+    },
+}
+
+class ArangoDBIntegration {
+    constructor(config) {
+        config.auth = {
+            username: config.username,
+            password: config.password,
+        }
+        
+        this.config = config
+        this.client = new Database(config)
+    }
+
+    async read(query) {
+        try {
+            const result = await this.client.query(query.sql)
+            let rl = []
+            await result.forEach((r)=> rl.push(r))
+            return rl
```

Comment:
You should be able to just `return result` here for the same effect.

Reply:
I tried this:
```js
  async read(query) {
    try {
      const result = await this.client.query(query.sql)
      return result
    } catch (err) {
      console.error("Error querying arangodb", err.message)
      throw err
    } finally {
      this.client.close()
    }
  }
```

but errors:

```
{
@budibase/server:       "type": "TypeError",
@budibase/server:       "message": "Converting circular structure to JSON\n    --> starting at object with constructor 'LinkedListItem'\n    |     property 'before' -> object with constructor 'LinkedListItem'\n    --- property 'behind' closes the circle",
@budibase/server:       "stack":
@budibase/server:           TypeError: Converting circular structure to JSON
@budibase/server:               --> starting at object with constructor 'LinkedListItem'
@budibase/server:               |     property 'before' -> object with constructor 'LinkedListItem'
@budibase/server:               --- property 'behind' closes the circle
@budibase/server:               at JSON.stringify (<anonymous>)
@budibase/server:               at Object.get length [as length] (/home/sovlookup/桌面/budibase/test/budibase/packages/server/node_modules/_koa@2.13.1@koa/lib/response.js:213:35)
@budibase/server:               at /home/sovlookup/桌面/budibase/test/budibase/packages/server/node_modules/_koa-compress@4.0.1@koa-compress/lib/index.js:58:35
@budibase/server:               at processTicksAndRejections (node:internal/process/task_queues:94:5)
@budibase/server:     
}
```

it seems that this `result`'s ArrayCursor type cannot be stringified

Reply:
solved! thanks~🥳